api/src/main/java/org/openmrs/api/DiagnosisService.java

```diff
@@ -67,12 +68,40 @@
 	@Authorized({ PrivilegeConstants.GET_DIAGNOSES })
 	List<Diagnosis> getDiagnoses(Patient patient, Date fromDate);
 
+	/**
+	 * Gets diagnoses for an Encounter. When specified, this method only returns
+	 * primary or confirmed diagnoses.
+	 *
+	 * @param encounter the encounter for which to fetch diagnoses
+	 * @param primaryOnly whether to return only primary diagnoses
+	 * @param confirmedOnly whether to return only confirmed diagnoses
+	 * @return the list of (primary, confirmed) diagnoses for the given encounter
+	 */
+	@Authorized({ PrivilegeConstants.GET_DIAGNOSES })
+	List<Diagnosis> getDiagnosesForEncounter(Encounter encounter, boolean primaryOnly, boolean confirmedOnly);
```

Comment:
Our convention has been `getDiagnosesByEncounter`. That is, `By` instead of `For`